docs/docs/getting-started/developing-on-hydra.md

```diff
@@ -0,0 +1,89 @@
+  -- Set a timeout for the whole test to fail
+  failAfter 60 $
+    -- We create a temporary directory to contain all the files required to spin up a node
+    withTempDir "cardano-node-tmp-dir" $ \cTmpDir -> do
+      config <- newNodeConfig cTmpDir
+      -- This starts the BFTNode as described above
+      withBFTNode (contramap FromCardanoNode tracer) config $ \(RunningNode _ nodeSocket) -> do
+        -- We create a temporary directory to contain all the files required to spin up the hydra nodes
+        withTempDir "hydra-nodes-tmp-dir" $ \hTmpDir -> do
+          -- We generate pairs of cardano keys for alice and bob, this pair is a tuple of verification and signing key
+          aliceKeys <- generate genKeyPair
+          bobKeys <- generate genKeyPair
+
+          -- Here we generate signing keys for hydra, note these are different from the cardano skeys generated above
+          let aliceSk = generateSigningKey "alice"
+              bobSk = generateSigningKey "bob"
+
+          let cardanoKeys = [aliceKeys, bobKeys]
+              hydraKeys = [aliceSk, bobSk]
+
+          -- Hydra nodes take an id as one of their arguments, it is used to identify them with the head protocol
+          let firstNodeId = 0
+
+          -- This will spin up `n` hydra nodes, where `n` is the length of the cardanoKeys list
+          -- (which needs to match the length of hydraKeys)
+          withHydraCluster tracer hTmpDir nodeSocket firstNodeId cardanoKeys hydraKeys $ \nodes -> do
+            ...
+```
+
+At this point, we are ready to start sending commands to the hydra-nodes programmatically.
+Hydra-cluster also exposes a `send` function, which takes an instance of the hydra-node and uses its websocket connection to send commands to the process.
+Remember that we must handle the whole head initialisation inside the test as well, so the first step will always be for one of the two nodes to `send` an `Init` command
+to start the opening of the head.
+
+Finally, hydra-node exposes several useful functions to wait for output from the hydra nodes themselves, both these functions have a timeout so they will function as assertions from a testing point of view; that is, if the node does not output what we expect within a certain timeframe, an error will be thrown and the whole test will fail.
+
+Remember the nodes will always produce output in JSON format, so all the assertions will expect some form of Aeson.Value to check against.
+
+`waitFor` will take a list of hydra-nodes and an Aeson Value, and check if they all produce the expected output within the allowed timeframe.
+
+`waitMatch` instead can be used to parse the output and possibly extract some information from that to be used in subsequent steps of the test.
+For example, after we submit a transaction to the node, we can use `waitMatch` to parse the `SnaphotConfirmed` event and extract the transactions present in that snapshot to check if it contains the transaction we just sent.
```

Comment:
Instead of documenting the functions here, probably better to just link the haddocks (https://hydra.family/head-protocol/haddock/hydra-cluster/HydraNode.html) and enumerate them as handy functions. 

Should there be no haddocks -> move these docs there :)